vmm/src/lib.rs

```diff
@@ -1558,23 +1558,25 @@ impl RequestHandler for Vmm {
 
     fn vm_info(&self) -> result::Result<VmInfoResponse, VmError> {
         match &self.vm_config {
-            Some(config) => {
+            Some(vm_config) => {
                 let state = match &self.vm {
                     Some(vm) => vm.get_state()?,
                     None => VmState::Created,
                 };
+                let config = vm_config.lock().unwrap().clone();
```

Comment:
Not sure this is better. Before we cloned the `Arc` i.e. just gave us a new ref count we can use. Here we do a full clone of the `VmConfig` only to then throw it away after it's been serialised by the API call output. That seems less efficient than taking the `Mutex`.